### PR TITLE
Copy the CONTRIBUTING from coc.nvim.

### LINKED ISSUE #45: Should have a CONTRIBUTING.md file
**Summary of the new feature**

I spent an hour and a half on stream trying to get coc.nvim to load the local repository instead of pulling from npm. It would probably be a good idea to have a CONTRIBUTING.md file that can assist with a basic setup to allow local modifications of the file (the fun part for me was figuring out that I needed to run `build.ps1` instead of the `yarn install` command recommended by coc.nvim.

I think taking a page from [coc.nvim's file](https://github.com/neoclide/coc.nvim/blob/master/CONTRIBUTING.md) might be a good place to start, although it might also be overkill.

### PR DESCRIPTION
**Pull request by corbob**
coc.nvim is MIT licensed, has a reasonable document that covers most of our setup.
Changed it slightly to make it more PowerShell heavy.

Fixes #45

### CODE REVIEW

**Comment by corbob** [comment]
I probably should have marked this as WIP. I would like to cleanup a bit to make it more coc-powershell centric.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+Code contributions of just about any size are acceptable!
+
+The main difference between code contributions and documentation contributions is that contributing code requires inclusion of relevant tests for the code being added or changed. Contributions without accompanying tests will be held off until a test is added, unless the maintainers consider the specific tests to be either impossible, or way too much of a burden for such a contribution.
+
+To contribute code:
+
+- [Set up the project](#project-setup).
+- Make any necessary changes to the source code.
+- Include any [additional documentation](#contribute-documentation) the changes might need.
+- Make sure the code doesn't have lint issue by command `yarn lint` in your
```
Comment (TylerLeonhardt) [inline CONTRIBUTING.md]: we don’t use yarn

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  terminal.
+- Write tests that verify that your contribution works as expected when necessary.
+- Make sure all tests passed by command `yarn jest` in your terminal.
```
Comment (TylerLeonhardt) [inline CONTRIBUTING.md]: we don’t use yarn

**Comment by corbob** [comment]
I've updated with the finds from @TylerLeonhardt .

One thing that might be helpful is if we could somehow have a docker container to assist with development. Working on the PS Conference Book Volume 2, they've added a dev container that has all of the linting tools and such that can be spun up to do the tests. Would be good if we could get something similar to possibly reduce the barrier to entry for new contributors.

Once #49 is merged, perhaps we can build off of it for a development container 🤷‍♂

**Comment by TylerLeonhardt** [comment]
Looks good. The only feedback is that the `Contribute Code` section is a bit strict. Maybe tone down the bullet points? There’s something about comment messages and I don’t think we care that much.

**Comment by TylerLeonhardt** [comment]
Also mention we borrowed this from coc.nvim

**Comment by corbob** [comment]
> 
> 
> Looks good. The only feedback is that the `Contribute Code` section is a bit strict. Maybe tone down the bullet points? There’s something about comment messages and I don’t think we care that much.

Yeah, I saw it, but didn't get to actually looking at it. I understand the reasoning for it, but as you said, we don't seem to care that much about it. 



> Also mention we borrowed this from coc.nvim

I added a little tongue in cheek note that we shamelessly stole it from coc.nvim.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+- [Install npm](https://npmjs.com)
+- [Fork the project](https://guides.github.com/activities/forking/#fork)
+
+Then in your (PowerShell) terminal:
+
+- Add coc-powershell to your vim's rtp by `set runtimepath^=/path/to/coc-powershell`
+- `cd path/to/your/coc-powershell`
+- `./build.ps1`
+
+### Run Project Within a Docker Container
```
Comment (TylerLeonhardt) [inline CONTRIBUTING.md]: Let's leave this out for now since the Dockerfile isn't quite ready yet.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+- Go to https://github.com/yatli/coc-powershell/pulls and open a new pull request with your changes.
+- If your PR is connected to an open issue, add a line in your PR's description that says `Fixes: #123`, where `#123` is the number of the issue you're fixing.
+
+## Contribute Code
+
+We like code commits a lot! They're super handy, and they keep the project going and doing the work it needs to do to be useful to others.
+
+Code contributions of just about any size are acceptable!
+
+The main difference between code contributions and documentation contributions is that contributing code requires inclusion of relevant tests for the code being added or changed. Contributions without accompanying tests will be held off until a test is added, unless the maintainers consider the specific tests to be either impossible, or way too much of a burden for such a contribution.
```
Comment (yatli) [inline CONTRIBUTING.md]: We don't have the test pipeline yet 😜

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+If you want to go the usual route and run the project locally, though:
+
+- [Install PowerShell](https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell)
+- [Install Node.js](https://nodejs.org/en/download/)
+- [Install npm](https://npmjs.com)
+- [Fork the project](https://guides.github.com/activities/forking/#fork)
+
+Then in your (PowerShell) terminal:
+
+- Add coc-powershell to your vim's rtp by `set runtimepath^=/path/to/coc-powershell`
```
Comment (yatli) [inline CONTRIBUTING.md]: "Edit your vimrc, and ..."

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+- `cd path/to/your/coc-powershell`
+- `./build.ps1`
+
```
Comment (yatli) [inline CONTRIBUTING.md]: "then, start vim/neovim. the local instance of `coc-powershell` will be picked up automatically."
"you can check whether it's loaded by running the command `:CocList extensions` -- a local plugin will be listed with an `[RTP]` tag"
"note, if you have installed `coc-powershell` (managed by `coc.nvim`), you may need to uninstall that first"

File: CONTRIBUTING.md (hunk 4 above)
Comment (corbob) [reply CONTRIBUTING.md]: I know, and I thought about taking it out... but then I thought perhaps leaving it in and encouraging us to build tests from the get go would be good...

🤔 perhaps rewording it to state that tests are encouraged...

And even better would be adding a section on Contributing Tests...

File: CONTRIBUTING.md (hunk 3 above)
Comment (corbob) [reply CONTRIBUTING.md]: Oh I suppose, you know that was like 30 seconds of typing you're asking me to undo... 😂

File: CONTRIBUTING.md (hunk 4 above)
Comment (corbob) [reply CONTRIBUTING.md]: I took out the heavy requirement, instead suggesting that they're encouraged. Perhaps when we have a test pipeline we can add in a part about contributing tests.

Comment (yatli) [reply CONTRIBUTING.md]: yeah that would be enough for now;

File: CONTRIBUTING.md (hunk 6 above)
Comment (yatli) [reply CONTRIBUTING.md]: @corbob you say it better :)
btw, off topic: when writing papers the reviewers often criticize our work for "choppy writing", "hard to understand" etc. etc. -- and I think I still have a long way to go, learn from you guys, and try to understand the subtle differences between me and a native speaker.

Comment (TylerLeonhardt) [reply CONTRIBUTING.md]: You do a great job @yatli 😊

Comment (corbob) [reply CONTRIBUTING.md]: I agree with Tyler. I simply took what you gave me and spun it into a step by step list of what to do.

It is very much an iterative process. I don't know how many times I've gone back to something I've written and feel that it's really choppy (as in doesn't seem to flow smoothly from topic to topic).

**Pull request merged**